pkg/appregistry/appregistry.go

```diff
@@ -100,3 +107,23 @@ func NewClient(kubeconfig string, logger *logrus.Entry) (clientset marketplace.I
 	clientset, err = marketplace.NewForConfig(config)
 	return
 }
+
+func NewSecretClient(kubeconfig string, logger *logrus.Entry) (clientset *kubernetes.Clientset, err error) {
```

Comment:
nit: maybe `kubeclient` is a more appropriate name?

Reply:
Sure, updated in latest commit. I also updated the field name in the downloader to match